### PR TITLE
Tuck the language toolbar into sub menus so smaller screens can access

### PR DESCRIPTION
**Pull request by doctormo**
On inkscape.org we have many languages active at the same time. The menu for language is very long and often can not fit on a screen 900px high. This patch tucks away each of the functions into it's own sub-menu with the break appearing if any of the menus will be there.

### CODE REVIEW

**Comment by yakky** [comment]
@doctormo nice! Could you rebase on latest develop to let a few test fixes in?

**Comment by doctormo** [comment]
I've merged but not sure about githubs process, do I make a new merge request?

**Comment by yakky** [comment]
@doctormo nope, this is fine. Due to the changed menu layout, a test fails, though. Could you check here https://travis-ci.org/divio/django-cms/jobs/106064604#L3651 ?

**Comment by Moini** [comment]
(this will fix https://github.com/divio/django-cms/issues/3742 )

**Comment by doctormo** [comment]
Updated with a fixed test and a new test. It wasn't clear how to write a test for the copy plugin menu.

**Comment by yakky** [comment]
@doctormo what's unclear about the test for copy plugins menu?
You can:
 * create english and german titles
 * add a text plugin to the english page
 * generate a request in germa
 * assert the correct menu items exist

**Comment by yakky** [comment]
(also, check the leftover import to make code check happy)

File: cms/cms_toolbars.py [hunk 1]
```diff
@@ -421,30 +421,32 @@ def change_language_menu(self):
             remove = [(code, languages.get(code, code)) for code in self.page.get_languages() if code in languages]
             add = [l for l in languages.items() if l not in remove]
             copy = [(code, name) for code, name in languages.items() if code != self.current_lang and (code, name) in remove]
-            if add:
+
+            if add or remove or copy:
                 language_menu.add_break(ADD_PAGE_LANGUAGE_BREAK)
+
+            if add:
+                add_plugins_menu = language_menu.get_or_create_menu(LANGUAGE_MENU_IDENTIFIER+'-add', _('Add Translation'))
```
Comment (yakky) [inline cms/cms_toolbars.py]: Better use ``format`` here instead of mere concatenation (same for the other menus)

**Comment by doctormo** [comment]
@yakky the tests are very very wet. That is to say, they aren't DRY. Developing assert base classes can reduce the line count in tests and increase maintainability of testing code by multiple factors saving developer time.

I'm going to let you fix the code. I'm not pleased with django-cms developers at the moment and I just don't have the patience for fiddling with code standards when the rest of the code isn't held to the same quality control.

**Comment by yakky** [comment]
Wet tests are much valued in django CMS test base, as give a much wider path coverage. But it's ok.
I can take care myself of getting this into shape

**Comment by yakky** [comment]
@mkoistinen @FinalAngel @vxsx @czpython what do you think of this PR? TL;DR: Changing "add translation", "delete translation", "copy plugins" into submenus with a list of languages.
Looks nice to me

**Comment by vxsx** [comment]
I'm all :+1: on it

**Comment by doctormo** [comment]
Thanks @yakky +1

**Comment by yakky** [comment]
Thanks @doctormo for the suggestion

**Pull request closed without merging**